ramlfications/utils/types.py

```diff
@@ -63,4 +66,64 @@ def parse_type(name, raw, root):
             data.pop("properties")
         except KeyError:
             pass
+
+    data.update(parse_examples(root.raml_version, data))
+
     return data_type_cls(**data)
+
+
+def parse_examples(raml_version, data):
+    #
+    # This is the only place that knows when to supply examples and when
+    # not, and that checks when both are supplied.
+    #
+    # This is also responsible for generating structured Example objects
+    # when appropriate (RAML >= 1.0).
+    #
+    data = data or {}
+    kw = {}
+    example = data.get("example", None)
+
+    if raml_version == "0.8":
+        if "examples" in data:
+            del data["examples"]
+            # Should emit a lint warning if there's an examples node and
+            # we're processing RAML 0.8: authors may be expecting it to
+            # be honored.
+        kw["example"] = example
+
+    if raml_version >= "1.0":
+        if "example" in data:
+            kw["example"] = parse_example(None, example)
+
+        if "examples" in data:
+            if "example" in data:
+                raise RuntimeError("example and examples cannot co-exist")
+            examples = data["examples"]
+            # Must be a map:
+            if not isinstance(examples, dict):
+                # Need to decide what exception to make this.
+                raise UnknownDataTypeError
+            kw["examples"] = NodeList([parse_example(nm, val)
+                                       for nm, val in iteritems(examples)])
+        else:
+            kw["examples"] = None
+
+    return kw
+
+
+def parse_example(name, node):
+    data = dict(name=name, value=node)
+    if name:
+        data["display_name"] = BaseContent(name)
```

Comment:
`BaseContent` is meant only for `root.documentation` and for descriptions (basically wherever there could be markdown). I've just set `name` to `display_name` if `displayName` isn't set.